### PR TITLE
Use `format_type` for `msg.type_not_iterable`

### PR DESCRIPTION
**Pull request by 97littleleaf11**
### Description

Related https://github.com/python/mypy/issues/5072

Uses `format_type` to get a more concise output.

### CODE REVIEW

File: mypy/messages.py [hunk 1]
```diff
@@ -1613,8 +1613,8 @@ def generate_incompatible_tuple_error(self,
         for i, (lhs_t, rhs_t) in enumerate(zip(lhs_types, rhs_types)):
             if not is_subtype(lhs_t, rhs_t):
                 if error_cnt < 3:
-                    notes.append('Expression tuple item {} has type "{}"; "{}" expected; '
-                        .format(str(i), format_type_bare(rhs_t), format_type_bare(lhs_t)))
+                    notes.append('Expression tuple item {} has type {}; {} expected; '
+                        .format(str(i), format_type(rhs_t), format_type(lhs_t)))
                 error_cnt += 1
```
Comment (97littleleaf11) [inline mypy/messages.py:1618]: Changing from `format_type_bare` to `format_type` to keep consistency with other usage of `format_type`.

File: test-data/unit/check-tuples.test [hunk 2]
```diff
@@ -1468,7 +1468,7 @@ x9, y9, x10, y10, z5 = *points2, 1, *points2 # E: Contiguous iterable with same
 () = []  # E: can't assign to ()
 
 [case testAssignEmptyBogus]
-() = 1  # E: "Literal[1]?" object is not iterable
```
Comment (sobolevn) [inline test-data/unit/check-tuples.test:1471]: This is a significant change and I don't think that is an improvement. `Literal[1]` is more spefic, I personally like it more.

File: test-data/unit/check-unions.test [hunk 3]
```diff
@@ -556,7 +556,7 @@ reveal_type(d1)  # N: Revealed type is "Union[Any, builtins.float]"
 reveal_type(d2)  # N: Revealed type is "Union[Any, builtins.float]"
 
 e: Union[Any, Tuple[float, float], int]
-(e1, e2) = e  # E: "builtins.int" object is not iterable
```
Comment (sobolevn) [inline test-data/unit/check-unions.test:559]: There are a lot of places where `builtins.` is used, and some places where it is not used. I cannot tell what kind of principle is used here 😞

**Review by sobolevn — COMMENTED** [review]
Thanks! I have a couple of questions 🙂

File: test-data/unit/check-unions.test (hunk 3 above)
Comment (97littleleaf11) [reply test-data/unit/check-unions.test:559]: `mypy` keeps the prefix in `reveal_type` for more verbose information. And according to the original discussion, it's better to discard `builtins.` to get concise error messages. Also, cpython dosen't output `builtins.` when error occurs.

File: test-data/unit/check-tuples.test (hunk 2 above)
Comment (97littleleaf11) [reply test-data/unit/check-tuples.test:1471]: yeah, this is an interesting case. I think it's a different issue that we might need to modify the `format_type`.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**